aviary/mission/gasp_based/ode/taxi_eom.py

```diff
@@ -11,37 +12,45 @@ class TaxiFuelComponent(om.ExplicitComponent):
     """
 
     def initialize(self):
+        self.options.declare('num_nodes', default=1, types=int)
         self.options.declare(
             'aviary_options', types=AviaryValues,
             desc='collection of Aircraft/Mission specific options')
 
     def setup(self):
+        nn = self.options['num_nodes']
+
         self.add_input(
             Dynamic.Mission.FUEL_FLOW_RATE_NEGATIVE_TOTAL,
-            val=1.0,
+            val=np.ones(nn),
             units="lbm/s",
             desc="fuel flow rate",
         )
         add_aviary_input(self, Mission.Summary.GROSS_MASS, val=175400.0)
 
         self.add_output(
             "taxi_fuel_consumed",
-            val=1.0,
+            val=np.ones(nn),
             units="lbm",
             desc="taxi_fuel_consumed",
         )
         self.add_output(
             Dynamic.Mission.MASS,
-            val=175000.0,
+            val=np.ones(nn)*175000.0,
             units="lbm",
             desc="mass after taxi",
         )
 
+    def setup_partials(self):
+        arange = np.arange(self.options['num_nodes'])
```

Comment:
```suggestion
        nn = self.options['num_nodes']
        arange = np.arange(nn)
```

Reply:
Doing this so nn is available later for another change

Reply:
Agree. Done.